scipy/signal/ltisys.py

```diff
@@ -2494,3 +2496,218 @@ def place_poles(A, B, poles, method="YT", rtol=1e-3, maxiter=30):
     full_state_feedback.nb_iter = nb_iter
 
     return full_state_feedback
+
+
+def controller_Hessenberg(A,B, C=None, tol=None):
+    r"""
+    Reduce the linear system ``AX + BU`` to the controller Hessenberg form
+    H, B_bar where P*A*P.T = H and P*B = Bbar with P being an orthogonnal
+    similarity transformation.
+
+    The new system has the following form:
+    
+    .. math:: 
+
+        H = \left [ \begin{array}{cc}
+                        H_c & H_{xx} \\
+                        0 & H_{nc}
+                        \end{array} \right ]
+
+    H is a block upper Hessenberg matrix.
+    
+    .. math:: 
+
+        \bar B = \left [ \begin{array}{c} B_c \\ 0 \end{array} \right ]
+
+        
+    Hc,Bc is the controllable part of the original system and ``AX + BU`` is
+    controllable if H is of full rank. If the real part of the eigenvalues of
+    Hnc are negative the system will be stabilizable.
+        
+    Parameters
+    ----------
+    A, B : ndarray
+        State-space representation of linear system ``AX + BU``.
```

Comment:
Iirc it was OK in place_poles.  http://docs.scipy.org/doc/scipy-0.16.1/reference/generated/scipy.signal.place_poles.html